### PR TITLE
Batch pruning of slabs after deleting objects from the database

### PR DESCRIPTION
**Pull request by ChrisSchinnerl**
This speeds up deletions and avoids redundant deletions in scenarios where files get deleted rapidly. e.g. when running `rclone sync` which might replace lots of files one-by-one.

It also puts the pruning of slabs into a background goroutine. There is no reason for an object deletion to fail when pruning slabs fails. This might lead to brief inconsistencies on the object stats endpoint but will greatly improve the speed at which we can delete objects and also reduce the database locking necessary when pruning slabs.

Closes https://github.com/SiaFoundation/renterd/issues/1149

### CODE REVIEW

File: stores/metadata.go [hunk 1]
```diff
@@ -2713,8 +2719,36 @@ func archiveContracts(tx *gorm.DB, contracts []dbContract, toArchive map[types.F
...
 
+func (s *SQLStore) pruneSlabsLoop() {
+	for {
+		select {
+		case <-s.slabPruneSigChan:
+		case <-s.shutdownCtx.Done():
+			return
+		}
+
+		ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
```
Comment (peterjan) [inline stores/metadata.go:2730]: this seems like an awfully tight timeout for a retry loop

File: stores/metadata_test.go [hunk 2]
```diff
@@ -4097,26 +4097,34 @@ func TestSlabCleanupTrigger(t *testing.T) {
...
+		}
+		return nil
+	})
 
 	// delete second object
 	err = ss.RemoveObject(context.Background(), api.DefaultBucketName, obj2.ObjectID)
 	if err != nil {
 		t.Fatal(err)
 	}
+	time.Sleep(100 * time.Millisecond)
```
Comment (peterjan) [inline stores/metadata_test.go:4118]: same here

File: stores/metadata_test.go [hunk 3]
```diff
@@ -4156,11 +4164,16 @@ func TestSlabCleanupTrigger(t *testing.T) {
 	if err != nil {
 		t.Fatal(err)
 	}
-	if err := ss.db.Model(&dbSlab{}).Count(&slabCntr).Error; err != nil {
-		t.Fatal(err)
-	} else if slabCntr != 1 {
-		t.Fatalf("expected 1 slabs, got %v", slabCntr)
-	}
+	time.Sleep(100 * time.Millisecond)
```
Comment (peterjan) [inline stores/metadata_test.go:4167]: and here

File: stores/metadata.go (hunk 1 above)
Comment (ChrisSchinnerl) [reply stores/metadata.go]: It should more than cover the sum of all retry intervals. Changed it a bit to make it more obvious.

**Review by peterjan — APPROVED** [review]

**Pull request merged**